synapse_spamcheck_badlist/bad_list_filter.py

```diff
@@ -127,17 +130,19 @@ async def check_event_for_spam(self, event) -> bool:
             # Check for links in text, both unformatted and formatted.
             #
             # We always lower-case the url, as the IWF database is lowercase.
-            for text in [content.get("body", ""), content.get("formatted_body", "")]:
-                # Run a first, faster test.
-                if not self._linkifier.test(text):
-                    continue
-                # Now run the slower test, if necessary, using results cached from the faster test.
-                for match in self._linkifier.match(text) or []:
-                    link = re.sub(self._scheme_re, "", match.url.lower())
-                    is_bad_link = await self._api.run_db_interaction("Check link against evil db", _db_is_bad_link, self._links_table, link)
-                    if is_bad_link:
-                        logger.info("Rejected bad link")
-                        return True
+            with self._link_test_performance.time():
+                for text in [content.get("body", ""), content.get("formatted_body", "")]:
+                    # Run a first, faster test.
+                    with self._linkify_test_performance.time():
```

Comment:
Do you want to tag this with whether it is body vs. formatted body?

Reply:
I wasn't planning to, but I can if you think it will be useful.

Reply:
🤷 I'm not sure if it would be useful or not, could always be added after.